playbooks/common/openshift-cluster/validate_hostnames.yml

```diff
@@ -0,0 +1,28 @@
+---
+- include: evaluate_groups.yml
+
+- name: Gather and set facts for node hosts
+  hosts: oo_nodes_to_config
+  roles:
+  - openshift_facts
+  tasks:
+  - openshift_facts:
+      role: "{{ item.role }}"
+      local_facts: "{{ item.local_facts }}"
+    with_items:
+      - role: common
+        local_facts:
+          hostname: "{{ openshift_hostname | default(None) }}"
+          public_hostname: "{{ openshift_public_hostname | default(None) }}"
+  - shell:
+      getent ahostsv4 {{ openshift.common.hostname }} | head -n 1 | awk '{ print $1 }'
```

Comment:
I couldn't come up with a better solution to cover cases where they had /etc/hosts entries, though now that I think of it if they do have /etc/hosts entries those wouldn't be reflected in containerized installs.